### PR TITLE
feat: support databend for superset

### PR DESCRIPTION
**Pull request by hantmac**
### SUMMARY

Support new database - [databend](https://github.com/datafuselabs/databend) for superset

### TESTING INSTRUCTIONS

pytest tests/unit_tests/db_engine_specs/test_databend.py

### CODE REVIEW

File: tests/unit_tests/db_engine_specs/test_databend.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        ("DateTime", "to_dateTime('2019-01-02 03:04:05')"),
+        ("UnknownType", None),
+    ],
+)
+def test_connect_convert_dttm(
+    target_type: str, expected_result: Optional[str], dttm: datetime
+) -> None:
+    from superset.db_engine_specs.databend import DatabendEngineSpec as spec
+
+    assert_convert_dttm(spec, target_type, expected_result, dttm)
```
Comment (villebro) [inline tests/unit_tests/db_engine_specs/test_databend.py:85]: This looks like a duplicate of `test_convert_dttm` on line 50?

File: tests/unit_tests/db_engine_specs/test_databend.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        ("Decimal256(2)", DECIMAL, None, GenericDataType.NUMERIC, False),
+        ("Bool", Boolean, None, GenericDataType.BOOLEAN, False),
+        ("Nullable(Bool)", Boolean, None, GenericDataType.BOOLEAN, False),
+        ("Date", Date, None, GenericDataType.TEMPORAL, True),
+        ("Nullable(Date)", Date, None, GenericDataType.TEMPORAL, True),
+        ("Datetime", DateTime, None, GenericDataType.TEMPORAL, True),
+        ("Nullable(Datetime)", DateTime, None, GenericDataType.TEMPORAL, True),
+    ],
+)
+def test_connect_get_column_spec(
```
Comment (villebro) [inline tests/unit_tests/db_engine_specs/test_databend.py:122]: Not that it matters that much, but just curious why this has `connect` in the function name?

File: tests/unit_tests/db_engine_specs/test_databend.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+
+@pytest.mark.parametrize(
+    "column_name,expected_result",
+    [
+        ("time", "time_07cc69"),
+        ("count", "count_e2942a"),
+    ],
+)
+def test_connect_make_label_compatible(column_name: str, expected_result: str) -> None:
```
Comment (villebro) [inline tests/unit_tests/db_engine_specs/test_databend.py:141]: Same here

**Review by villebro — COMMENTED** [review]
Thanks for adding support for Databend! A few minor comments, other than that LGTM 👍

File: tests/unit_tests/db_engine_specs/test_databend.py (hunk 1 above)
Comment (hantmac) [reply tests/unit_tests/db_engine_specs/test_databend.py:85]: Aha, that's it. Thanks!

File: tests/unit_tests/db_engine_specs/test_databend.py (hunk 3 above)
Comment (hantmac) [reply tests/unit_tests/db_engine_specs/test_databend.py:141]: Fixed too.

**Comment by hantmac** [comment]
@villebro Fixed the comment and would like to help me fix this issue https://github.com/apache/superset/issues/23304?

**Comment by hantmac** [comment]
> Thanks for adding support for Databend! A few minor comments, other than that LGTM 👍

Hi @villebro , since this pr has been finished, how about merging this pr?

**Comment by villebro** [comment]
> > Thanks for adding support for Databend! A few minor comments, other than that LGTM 👍
> 
> Hi @villebro , since this pr has been finished, how about merging this pr?

Hi @hantmac - I'm so sorry for dropping the ball on this PR. I've merged master on it to restart CI, and will re-review and merge if everything is ok today

**Review by villebro — APPROVED** [review]
LGTM, thanks for the fantastic quality of this PR (tests, docs etc) ❤️

**Comment by villebro** [comment]
@hantmac it seems there's quite a few linting errors here:
image

Would you be able to resolve them? If not I can pick this up and push it through next week.

**Comment by hantmac** [comment]
> @hantmac it seems there's quite a few linting errors here: image
> 
> Would you be able to resolve them? If not I can pick this up and push it through next week.

Thanks for your review! I will fix it as soon as possible.

**Comment by villebro** [comment]
@hantmac I restarted CI, will take a look tomorrow morning PST

File: superset/db_engine_specs/databend.py [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+
+class DatabendBaseEngineSpec(BaseEngineSpec):
+    """Shared engine spec for Databend."""
+
+    time_secondary_columns = True
+    time_groupby_inline = True
+
+    _time_grain_expressions = {
+        None: "{col}",
+        "PT1M": "to_start_of_minute(TO_DATETIME({col}))",
```
Comment (michael-s-molina) [inline superset/db_engine_specs/databend.py:48]: Could you use `superset.constants.TimeGrain`?

`TimeGrain.MINUTE: "to_start_of_minute(TO_DATETIME({col}))",`

**Pull request merged**